gremlin-go/driver/graphTraversal.go

```diff
@@ -669,6 +670,38 @@ func (g *GraphTraversal) Write(args ...interface{}) *GraphTraversal {
 	return g
 }
 
+const maxInt32 = 2147483647
+
+func int32Args(args ...interface{}) []interface{} {
+	for i, arg := range args {
+		switch val := arg.(type) {
+		case uint:
+			if val <= maxInt32 {
+				args[i] = int32(val)
+			}
+		case uint32:
+			if val <= maxInt32 {
+				args[i] = int32(val)
+			}
+		case uint64:
+			if val <= maxInt32 {
+				args[i] = int32(val)
+			}
+		case int:
+			if val <= maxInt32 {
+				args[i] = int32(val)
+			}
+		case int64:
+			if val <= maxInt32 {
+				args[i] = int32(val)
+			}
```

Comment:
Does it make sense to also check for `val >= math.MinInt32` in the case of `int` and `int64`? I guess the server would fail anyway, but being strict we could be truncating big negative numbers.